Source/HYPScatterPlot.m

```diff
@@ -28,136 +35,212 @@ - (instancetype)initWithFrame:(CGRect)frame
     self = [super initWithFrame:frame];
     if (!self) return nil;
 
-    if (self.backgroundColor) {
-        self.backgroundColor = [UIColor colorFromHex:HYPScatterPlotBackgroundColor];
-    }
-
-    if (!self.xAxisColor) {
-        self.xAxisColor = [UIColor colorFromHex:HYPScatterPlotXLineColor];
-    }
-
-    if (!self.averageLineColor) {
-        self.averageLineColor = [UIColor whiteColor];
-    }
-
-    if (!self.yAxisMidGradient) {
-        self.yAxisMidGradient = [UIColor whiteColor];
-    }
-
-    if (!self.yAxisEndGradient) {
-        self.yAxisEndGradient = [UIColor colorFromHex:HYPScatterPlotBackgroundColor];
-    }
+    self.backgroundColor = [UIColor colorFromHex:HYPScatterPlotBackgroundColor];
+
+    self.averageLineColor = [UIColor whiteColor];
+    self.xAxisColor = [UIColor colorFromHex:HYPScatterPlotXLineColor];
+    self.yAxisMidGradient = [UIColor whiteColor];
+    self.yAxisEndGradient = [UIColor clearColor];
+    self.defaultPointFillColor = [UIColor whiteColor];
+    self.selectedPointFillColor = [UIColor lightGrayColor];
+    self.selectedPointStrokeColor = [UIColor whiteColor];
+    self.axisLineWidth = HYPScatterPlotAxisLineWidth;
+    self.pointRadius = HYPScatterPlotCircleRadius;
+
+    UIView *leftYAxis = [UIView new];
+    leftYAxis.translatesAutoresizingMaskIntoConstraints = NO;
+    self.leftYAxisGradient = [CAGradientLayer layer];
```

Comment:
Could you move initialization to a getter method? Also known as lazy initialization (which is a kind of standard here)? :)